docs/python-grader/index.md

```diff
@@ -26,31 +26,68 @@ A full `info.json` file should look something like:
 }
 ```
 
-### `server.py`
+### Variables for and from the user
 
-The server code in the `generate()` function must define the list of variables or functions that will be passed to the autograded student code as `names_for_user`, and also those that will be passed from the student code to the test code as `names_from_user`. Only variables or functions listed in `names_for_user` will be accessible by the user from the setup code; only names listed in `names_from_user` will be accessible by the test cases from the user code.
+The question must specify the variables that will be read from student code for grading (`names_from_user`). Optionally, the question may also specify variables that will be made available to student code from the question's setup code (`names_for_user`). Only variables or functions listed in `names_for_user` will be accessible by the user from the setup code; only names listed in `names_from_user` will be accessible by the test cases from the user code.
 
-These are stored as a list of dictionary objects in the `data["params"]` dict. The above `names_for_user` and `names_from_user` lists are stored as separate keys in `params`. For example:
+Each variable must have the following:
+
+- `name`: The name of the variable in the student code.
+- `description`: A human-readable description of the variable.
+- `type`: A human-readable description of the type of the variable. This is used for display purposes only, and does not affect the grading.
+
+There are two ways to do specify these variables.
+
+#### `question.html`
```

Comment:
nit: consider a tabbed interface with `question.html` and `server.py`.

Reply:
Done in 94a057e.